XMonad/Config/Mate.hs

```diff
@@ -31,7 +31,6 @@ module XMonad.Config.Mate (
 import XMonad
 import XMonad.Config.Desktop
 import XMonad.Util.Run (safeSpawn)
-import XMonad.Util.Ungrab
```

Comment:
This will result in a build failure related to [this](https://github.com/xmonad/xmonad-contrib/pull/843#discussion_r1396790444). I'm not sure it's worth bumping our bounds over, so we should probably be a bit more tame here as well

Reply:
What I found is that if I don't remove this line, the compiler will complain about an ambiguous reference in the new version. Because `unGrab` will be defined both in `XMonad.Operations` (which is imported into this file from `import XMonad`) and `XMonad.Util.Ungrab`.

But thinking about it again. I could probably just do `import XMonad hiding (unGrab)`. Would that be preferrable?

Reply:
Yes, I was thinking about hiding the function from the `XMonad` import and perhaps adding a `{-# OPTIONS_GHC -Wno-dodgy-imports #-}` at the top of the file, to pacify the compiler when this is compiled against xmonad HEAD

Reply:
I'd be in favor of bumping xmonad and having `#if MIN_VERSION_xmonad(0, 17, 3)` in X.U.Ungrab so that users following the stable releases don't get the `Ambiguous occurrence ‘unGrab’` error when compiling their configs, and just get a deprecaton warning.

Reply:
(Note that for that to work, we can't have `unGrab = Operations.unGrab` but must reexport the import.)

Reply:
Hm, thinking a bit more about this, might as well bump the dependency instead of introducing an `#if`… :-)

Reply:
> I'd be in favor of bumping xmonad and having `#if MIN_VERSION_xmonad(0, 17, 9)` in X.U.Ungrab so that users following the stable releases don't get the `Ambiguous occurrence ‘unGrab’` error when compiling their configs, and just get a deprecaton warning.

Right now users compiling against `xmonad-0.17.2` should not get an error (as evidenced by the CI), since we're hiding (the possible non-existent) `unGrab` from the `XMonad` import. Still, bumping might not be such a bad idea

Reply:
I wasn't talking about compiling xmonad-contrib itself, but about compiling the xmonad.hs config. Anyone who imports both `XMonad` and `XMonad.Util.Ungrab` will start getting errors. I mean, yeah, it is mentioned in "Breaking Changes", but I don't think it should actually be a _Breaking Change_ when we can turn it into just a deprecation warning.

Reply:
(https://github.com/xmonad/xmonad/commit/5c6c3d4d71dc28502ffbcf3a7fddad64c6b3d8b8 & https://github.com/liskin/xmonad-contrib/commit/8526d935066e7de5dd5cadd7afca208d7e11b754 are one way to solve this, if we want an ifdef)

Reply:
Oh, right, that might indeed be an issue. The ifdef sounds fine in that case

Reply:
Okay, I pushed the two commits then.